WordPress/gutenberg
# Site Editor: Resolve homepage template on server-side

## Description
Closes #36873.

PR introduces a new `gutenberg_resolve_home_template` function, which will try to resolve the home template on the server-side. The resolved template is available via `__unstableHomeTemplate` editor settings.

### Notes
* The redirection happens on the server-side.
* The `gutenberg_resolve_home_template` function doesn't use PHP template fallbacks. So if a theme has `front-page.php` and `index.html,` the editor will display `index.html` while the site displays `front-page.php.` Previously, this caused an error in the editor since we can't render PHP templates.

## Testing Instructions
Tested using TT2 theme.

* Confirm that the correct template is displayed when accessing Site Editor.
* Confirm that template hierarchy is respected by creating `front-page.html` in the templates directory.
* Check that the static homepage works from the Reading settings.

## Types of changes
Enhancement

## Checklist:
